lib/molinillo/dependency_graph.rb

```diff
@@ -56,18 +52,19 @@ def initialize
     # have the correct {Vertex#graph} set
     def initialize_copy(other)
       super
-      @vertices = other.vertices.reduce({}) do |vertices, (name, vertex)|
-        vertices.tap do |hash|
-          hash[name] = vertex.dup.tap { |v| v.graph = self }
+      @vertices = {}
+      @root_vertices = {}
+      traverse = lambda do |new_v, old_v|
+        new_v.explicit_requirements.replace old_v.explicit_requirements
+        return if new_v.outgoing_edges.size == old_v.outgoing_edges.size
+        old_v.outgoing_edges.each do |edge|
+          destination = add_vertex(edge.destination.name, edge.destination.payload)
+          add_edge_no_circular(new_v, destination, edge.requirement)
+          traverse.call(destination, edge.destination)
```

Comment:
the lambda is only ever defined once per copy, and the overhead has to be minimal compared to the work it's doing